core/Pos/Core/Block/Union/Instances.hs

```diff
@@ -71,6 +82,9 @@ instance Bi BlockHeader => IsHeader BlockHeader
 -- HasPrevBlock
 ----------------------------------------------------------------------------
 
+instance HasPrevBlock BlockHeader where
+    prevBlockL = choosingBlockHeader prevBlockL prevBlockL
```

Comment:
Is `choosingBlockHeader` is not like `either` for `BlockHeader`? If so, then why not use it everywhere?

Reply:
It's like `choosing` from `Lens`, a more powerful combinator.